packages/sign-client/src/controllers/engine.ts

```diff
@@ -856,6 +866,16 @@ export class Engine extends IEngine {
     }
     const { pairingTopic, requiredNamespaces, relays } = params;
     if (!isUndefined(pairingTopic)) await this.isValidPairingTopic(pairingTopic);
+
+    // validate required namespaces only if they are defined
+    if (
```

Comment:
Maybe I'm not seeing something, but couldn't we use `!isValidObject()` as above rather than inlining ad-hoc conditions here?